test/explicit-length-check.js

```diff
@@ -0,0 +1,65 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/explicit-length-check';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'explicit-length-check',
+	message: '`length` property should be compared to a value'
+}];
+
+ruleTester.run('explicit-length-check', rule, {
+	valid: [
+		'array.foo',
+		'array.length',
+		'array.length === 0',
+		'array.length !== 0',
+		'array.length > 0',
+		'if (array.foo) {}',
+		'if (length) {}',
+		'if ([].length > 0) {}',
+		'if ("".length > 0) {}',
+		'if (array.length === 0) {}',
+		'if (array.length !== 0) {}',
+		'if (array.length !== 0 && array[0] === 1) {}'
+	],
+	invalid: [
+		{
+			code: 'if ([].length) {}',
+			errors
```

Comment:
In other rules, we went around this problem using this kind of construct. Might be simpler to do it this way.
https://github.com/sindresorhus/eslint-plugin-xo/blob/master/test/catch-error-name.js#L11-L17